package.json

```diff
@@ -54,14 +57,16 @@
     "prettier:fix": "prettier --write --ignore-path .gitignore \"**/*.{ts*,js,css,md}\"",
     "lint:fix": "tsc && npm run prettier:fix && npm run eslint:fix",
     "build": "rimraf public/build && npm run generate-icons && cross-env NODE_ENV=production remix build --sourcemap",
+    "dev:build": "npm run generate-icons && cross-env NODE_ENV=development remix build",
     "dev:remix": "cross-env NODE_ENV=development remix watch",
     "dev:miniflare": "cross-env NODE_ENV=development miniflare ./build/index.js --watch",
-    "start": "npm run generate-icons && cross-env NODE_ENV=development remix build && run-p dev:*",
+    "start": "npm run dev:build && run-p dev:*",
```

Comment:
Ah yes, thanks for pointing this out.

So a challenge with this setup I've proposed in this PR is that the tests need to be run on javascript that compiled for the cloudflare workers (because the `testEnvironment` has changed from `node` to `miniflare`). The tests fail if there is no `build/index.js` file. This is why `npm run build` was added to the `pr-check.yml` GH action. 

So I think I was planning to change the `test` script in this file to be `npm run dev:build && jest`. Though maybe `npm run build && jest` would be just as good.
Probably this is suboptimal and it would better if jest could compile itself immediately prior to the test execution (which is what I think `ts-jest` does if running in `node` environment). So maybe there is a way to improve in this direction 🤔 

Any thoughts on this?

Reply:
not sure how to make it all work seamlessly, but `npm run build && jest` sounds OK

Reply:
Ok, I've changed it to be `npm run build && jest`. It does seem possible to get `ts-jest` to work with miniflare, though:
https://github.com/cloudflare/miniflare-typescript-esbuild-jest/commit/bc2391a03100a15330e0a2924926fc49c7b78ee5#diff-1e058ca1442e46581b13571fb8d261f9e1f5657e26c96634d4c1072f0f0347f1 
I will perhaps give this a go in a separate PR.

Reply:
Ah, no, `npm run build && jest` doesn't quite work with the current config because `npm run build` is compiling to `public/build` where as `jest` is looking for `build/index.js` (without the `public`). So I'll have to improve this in this PR somehow.

Reply:
I'm not sure what changed, but `npm run build` now also compiling to `./build`. So `npm run build && jest` is now working just fine.